tsdb/engine/tsm1/compact_test.go

```diff
@@ -4305,66 +4286,60 @@ func TestEnginePlanCompactions(t *testing.T) {
 				},
 				{
 					FileStat: tsm1.FileStat{
-						Path: "01-05.tsm",
+						Path: "12-05.tsm",
 						Size: 256 * 1024 * 1024,
 					},
 					FirstBlockCount: tsdb.DefaultAggressiveMaxPointsPerBlock,
 				},
 				{
 					FileStat: tsm1.FileStat{
-						Path: "02-05.tsm",
+						Path: "13-05.tsm",
 						Size: 256 * 1024 * 1024,
 					},
 					FirstBlockCount: tsdb.DefaultMaxPointsPerBlock,
 				},
 				{
 					FileStat: tsm1.FileStat{
-						Path: "03-05.tsm",
+						Path: "14-05.tsm",
 						Size: 256 * 1024 * 1024,
 					},
 					FirstBlockCount: tsdb.DefaultMaxPointsPerBlock,
 				},
 				{
 					FileStat: tsm1.FileStat{
-						Path: "04-04.tsm",
+						Path: "15-04.tsm",
 						Size: 256 * 1024 * 1024,
 					},
 					FirstBlockCount: tsdb.DefaultMaxPointsPerBlock,
 				},
 				{
 					FileStat: tsm1.FileStat{
-						Path: "12-01.tsm",
+						Path: "16-01.tsm",
 						Size: 256 * 1024 * 1024,
 					},
 					FirstBlockCount: 50,
 				},
 				{
 					FileStat: tsm1.FileStat{
-						Path: "13-01.tsm",
+						Path: "17-01.tsm",
 						Size: 256 * 1024 * 1024,
 					},
 					FirstBlockCount: 50,
 				},
 				{
 					FileStat: tsm1.FileStat{
-						Path: "14-01.tsm",
+						Path: "18-01.tsm",
 						Size: 256 * 1024 * 1024,
 					},
 					FirstBlockCount: 50,
 				},
 			},
 			testShardTime: -1,
-			expectedResult: func() testLevelResults {
-				return testLevelResults{
-					level1Groups: []tsm1.PlannedCompactionGroup{
-						{
-							tsm1.CompactionGroup{"05-01.tsm", "06-01.tsm", "07-01.tsm", "08-01.tsm", "09-01.tsm", "10-01.tsm", "11-01.tsm", "12-01.tsm"},
```

Comment:
again, what is the intent of the original test. `Mock another planned level inside scheduler aggress blocks middle` is tough for me to understand, but you've significantly changed the expectations of this test so i think you've changed its intent too much. There's no level1 plan at all anymore!

I think you need to keep the tsm files the same names but properly sort them in `files: []tsm1.ExtFileStat{` test setup. That should keep the test expectations unchanged. Please make it so

Reply:
I agree.  Sort the input test files, but change nothing else.